### ISSUE TITLE
code review

### ISSUE THREAD

**Issue by chunhualiao** [bin123apple/Fortran2Cpp#57]

**Comment by chunhualiao** [comment]
```
You seem to be using the pipelines sequentially on GPU. In order to maximize efficiency please use a dataset
(.venv) [liaoch@nid200325:~/workspace-cfs/Fortran2Cpp/Evaluation]ls
CodeBLEU                   Groundtruth_C++.txt         text_generation_pipline_v2.py
compile_run_evaluation.py  text_generation_pipline.py  translation_log.txt

```

**Comment by bin123apple** [comment]
> You seem to be using the pipelines sequentially on GPU. In order to maximize efficiency please use a dataset

I think your suggestion is running several data samples at the same time. I will try to modify the code.

**Comment by chunhualiao** [comment]
yes, it takes a very long time to process ~300 inputs. Concurrent
processing may be helpful.

**Comment by chunhualiao** [comment]
There might be a step missing in README.md's "Reproduce Steps".
* step 2 generate the results, saved into the log file that has lots of mixed code with text. 
* step 3: calc_code_blue.py expects single line C++ programs to be compared with Groundtruth_C++.txt . 

What are the instructions to process the log file into the file acceptable by calc_code_blue.py? @bin123apple

**Comment by bin123apple** [comment]
I think this line : https://github.com/bin123apple/Fortran2Cpp/blob/main/Evaluation/text_generation_pipline.py#L58 should save all the code the log file acceptable by `calc_code_blue.py`
